include/v8.h

```diff
@@ -172,6 +172,21 @@ struct AutoDisallowEvents {
   ~AutoDisallowEvents() { EndDisallowEvents(); }
 };
 
+struct AutoMaybeDisallowEvents {
+  AutoMaybeDisallowEvents(bool disallowEvents, const char* label) {
+    if (disallowEvents) {
+      disallow = new AutoDisallowEvents(label);
```

Comment:
It would be better to use base::Optional to avoid the unnecessary heap allocation here.

Reply:
I tried that but in `v8.h`, I would not just want to add generic `include`s.

I'd love to just start migrating all our stuff from `v8.h` and `api.cc` to its own files. But to save myself sometime, I only want to start moving the new stuff to the new file.

Reply:
Will move it to a new file in `base/replayio.h`.

Reply:
OK if Optional<> isn't easy to use here the heap allocation is fine as is unless you really want to do this refactoring work.

Reply:
I realize you created the same thing and put it in `v8-runtime-agent-impl.cc` 😂